### DISCUSSION TITLE
Firmata WiFi/BLE

### DISCUSSION THREAD

**Discussion by djaus2** [earlephilhower/arduino-pico#1849] [Ideas]
Has anyone got Firmata to run on the RPi Pico W
I have got the WiFi version running on the device using[https://github.com/firmata/ConfigurableFirmata/tree/master](https://github.com/firmata/ConfigurableFirmata/tree/master) . Would like to get the BLE version from there to run.,  Can't get it to compile.
Also chasing a good text app. Have a C# console app that, acting as TCPIP Client can connect and communicate a bit but can't send commands

**Comment by djaus2** [comment]
After spending a day or so on this I've started creating a Firmata style package called Softatta from scratch. Not publicyet. Will comment here when available but will will include Softata app to run on RPi Pico W and use WiFi to connect witha a .NET library class.

**Comment by djaus2** [comment]
Have implemented Digital and Analog: https://github.com/djaus2/Soft-ata  _(Public now)_
- Includes C# Console Test app which can run in one of two modes: Digital or Analog.
- Arduino connectivity is over WiFi . 
- Functionality is in a C# class library which can be used elsewhere. Will publish on Nuget.
- Digital,Analog and Seril available.
- Blazor Client app as well.

**Reply by earlephilhower** [reply]
Great, thanks for letting us know!

**Comment by PizzaProgram** [comment]
Dear @djaus2 ,

As I see, after 1.5 years this is still in active development ! Cool.
- Do you plan to make it available for "simple RPi Pico" too? _(the non-Wireless version)_
- would You say it's **stable** now for production use? _(Especially disconnect/reconnect problems.)_

**Reply by earlephilhower** [reply]
I suggest posting something at the original repo in case he's not watching this one.  https://github.com/djaus2/Soft-ata

**Reply by PizzaProgram** [reply]
Thanks for the suggestion! I've [already done that](https://github.com/djaus2/Soft-ata/issues/1#issuecomment-2783595591).
